M1 builds broken (bitcoind dep not building for aarch64-apple-darwin)

**Describe the bug**  
bdk depends on electrsd crate which depends on bitcoind crate which wasn't building on M1. New version of bitcoind (0.21.0) is now out which fixes this (https://github.com/RCasatta/bitcoind/pull/38), but electrsd needs to update to it (issue opened https://github.com/RCasatta/electrsd/issues/25), and then bdk will need to update to that updated version of electrsd.

(I believe electrsd is the only reason bdk depends on bitcoind?)

**To Reproduce**  
running cargo test or any of the examples on M1 returns this error:

```
Compiling bitcoind v0.20.0
error[E0425]: cannot find function `download_filename` in this scope
  --> /Users/futurepaul/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoind-0.20.0/build.rs:49:29
   |
49 |     let download_filename = download_filename();
   |                             ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0277]: the size for values of type `str` cannot be known at compilation time
   --> /Users/futurepaul/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoind-0.20.0/build.rs:62:9
    |
62  | /         println!(
63  | |             "filename:{} version:{} hash:{}",
64  | |             download_filename, VERSION, expected_hash
65  | |         );
    | |__________^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `ArgumentV1::<'a>::new`
   --> /Users/futurepaul/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:293:20
    |
293 |     pub fn new<'b, T>(x: &'b T, f: fn(&T, &mut Formatter<'_>) -> Result) -> ArgumentV1<'b> {
    |                    ^ required by this bound in `ArgumentV1::<'a>::new`
    = note: this error originates in the macro `$crate::format_args_nl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the size for values of type `str` cannot be known at compilation time
   --> /Users/futurepaul/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoind-0.20.0/build.rs:67:19
    |
67  |           let url = format!(
    |  ___________________^
68  | |             "https://bitcoincore.org/bin/bitcoin-core-{}/{}",
69  | |             VERSION, download_filename
70  | |         );
    | |_________^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `ArgumentV1::<'a>::new`
   --> /Users/futurepaul/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:293:20
    |
293 |     pub fn new<'b, T>(x: &'b T, f: fn(&T, &mut Formatter<'_>) -> Result) -> ArgumentV1<'b> {
    |                    ^ required by this bound in `ArgumentV1::<'a>::new`
    = note: this error originates in the macro `$crate::__export::format_args` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0277, E0425.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `bitcoind` due to 3 previous errors
```

**Build environment**  
 - BDK tag/commit: `c1af45` (master) and `69b184a` (release/0.14.0)
 - OS+version: macOS 12.0.1, M1
 - Rust/Cargo version:  1.56.0
 - Rust/Cargo target: stable-aarch64-apple-darwin

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -94,7 +94,7 @@ test-md-docs = ["electrum"]
 lazy_static = "1.4"
 env_logger = "0.7"
 clap = "2.33"
-electrsd = { version= "0.12", features = ["trigger", "bitcoind_22_0"] }
+electrsd = { version= "0.13", features = ["trigger", "bitcoind_22_0"] }
 
 [[example]]
 name = "address_validator"

